playbooks/prepare_host.yaml

```diff
@@ -20,9 +20,27 @@
       key: "{{ item }}"
     loop: "{{ authorized_keys }}"
 
-  - name: Prepare host on RHEL system
+  - name: Prepare host on RHEL system with Subscription Manager
     when:
       - ansible_facts.distribution == 'RedHat'
+      - rhsm_enabled
+    block:
+      - name: Configure Red Hat Subscription Manager
+        import_role:
+          name: redhat-subscription
```

Comment:
I can't work out how this would authenticate to RHSM. We're talking about including subscription manager. Where would that go?

Or for the moment are we saying that you'd have to log in to RHSM manually on the host before executing the ansible? I'd be ok with that until we sort activation keys out, btw.

Reply:
This role will execute RHSM for you in the first thing, so later we can use DNF to pull rpms. I tested and it works.
As for activation keys, the user can define the Ansible vars for that role to use activation keys so no work will be needed to make it work with it.

Reply:
👆 is my remaining concern here.

I just asked @EmilienM how this works. The redhat-subscription role implicitly uses vars which must be defined in local-overrides. This isn't ideal (we should get activation keys working), but as these credentials are never persisted anywhere except the user's own workstation we could let it slide briefly for testing purposes. However, we *MUST* replace this with activation keys asap, and we *MUST NOT* integrate anything into CI which uses user credentials.